sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlInferSchema.scala

```diff
@@ -171,16 +171,18 @@ class XmlInferSchema(options: XmlOptions, caseSensitive: Boolean)
           case _: EndElement => StringType
           case _ => inferField(parser)
         }
-      case c: Characters if !c.isWhiteSpace =>
+      // what about new line character
+      case c: Characters if !isEmptyString(c) =>
```

Comment:
For this case, can't we return `inferObject(parser)`?
In `inferObject(parser)`, the case for `StructType` can be updated to "unnest" `StructType` with just `valueTag`.
Without this, there is lot of code duplication logic for valueTag.

Reply:
IMHO inferObject can't do this. This branch handles both primitive types and nested objects. If we return `inferObject(parser)`, the primitive types will be inferred as a structFields of valueTag